### PR TITLE
Add review rules regarding the owned files

### LINKED ISSUE #13791: [Meta] Reactivate `frame-coders` team
Reactivate the @paritytech/frame-coders team.  

Tasks:
- [ ] Update the member list
- [ ] Add folders to the CODEOWNERS file
- [ ] Add review rules regarding the owned files

### PR DESCRIPTION
**Pull request by juangirini**
Partially resolves #13791 

This PR adds a new Custom Review rule for code owned by the @paritytech/frame-coders team.

### CODE REVIEW

File: .github/pr-custom-review.yml [hunk 1]
```diff
@@ -8,12 +8,24 @@ rules:
...
+      # excluding files from 'CI team' and 'FRAME coders' rules
+      exclude: ^\.gitlab-ci\.yml|^scripts/ci/.*|^\.github/.*|^\.config/nextest.toml|^frame/(?!.*(nfts/.*|uniques/.*|babe/.*|grandpa/.*|beef|merkle-mountain-range/.*|contracts/.*|election|nomination-pools/.*|staking/.*))
     min_approvals: 2
     teams:
       - core-devs
 
+  - name: FRAME coders
+    check_type: changed_files
+    condition:
+      include: ^frame/(?!.*(nfts/.*|uniques/.*|babe/.*|grandpa/.*|beef|merkle-mountain-range/.*|contracts/.*|election|nomination-pools/.*|staking/.*))
```
Comment (juangirini) [inline .github/pr-custom-review.yml:20]: This line includes everything under `frame` expect for those named sub directories, or partially named ones (`beef` will match `beefy` and `beefy-mmr`)
@kianenigma @ggwpez do you think there's more we want to exclude?

File: .github/pr-custom-review.yml [hunk 2]
```diff
@@ -8,12 +8,24 @@ rules:
...
+      # excluding files from 'CI team' and 'FRAME coders' rules
+      exclude: ^\.gitlab-ci\.yml|^scripts/ci/.*|^\.github/.*|^\.config/nextest.toml|^frame/(?!.*(nfts/.*|uniques/.*|babe/.*|grandpa/.*|beefy|merkle-mountain-range/.*|contracts/.*|election|nomination-pools/.*|staking/.*))
     min_approvals: 2
     teams:
       - core-devs
 
+  - name: FRAME coders
+    check_type: changed_files
+    condition:
+      include: ^frame/(?!.*(nfts/.*|uniques/.*|babe/.*|grandpa/.*|beefy|merkle-mountain-range/.*|contracts/.*|election|nomination-pools/.*|staking/.*))
```
Comment (juangirini) [inline .github/pr-custom-review.yml:20]: This line includes everything under `frame` expect for those named sub directories, or partially named ones (`beef` will match `beefy` and `beefy-mmr`)
@kianenigma @ggwpez do you think there's more we want to exclude?

Comment (ggwpez) [reply .github/pr-custom-review.yml:20]: \+ aura

**Comment by juangirini** [comment]
bot rebase

**Review by alvicsam — APPROVED** [review]

**Review by rcny — APPROVED** [review]

**Comment by juangirini** [comment]
bot rebase

**Review by sam0x17 — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Pull request merged**